File: backend/routes/remediation.js
const express = require('express');
const router = express.Router();
const { generateRemediationScripts } = require('../services/remediation-service');

router.post('/generate', async (req, res) => {
    const { findings } = req.body;
    try {
        const scripts = await generateRemediationScripts(findings);
        res.json({ success: true, scripts });
    } catch (err) {
        res.status(500).json({ success: false, message: err.message });
    }
});

module.exports = router;

File: backend/routers/remediation.py
"""
Remediation router — script generation + AI chat (AI is ONLY used here).
"""

from fastapi import APIRouter
from pydantic import BaseModel
from typing import Optional

from services.remediation_service import generate_remediation_scripts
from services.ai_service import ask_remediation_expert

router = APIRouter()


class GenerateRequest(BaseModel):
    findings: list = []


class ChatRequest(BaseModel):
    message: str
    history: Optional[list] = []


@router.post("/generate")
def generate(body: GenerateRequest):
    scripts = generate_remediation_scripts(body.findings)
    return {"success": True, "scripts": scripts}


@router.post("/chat")
async def chat(body: ChatRequest):
    response = await ask_remediation_expert(body.message, body.history)
    return {"success": True, "response": response}
